src/gboml/semantic.py

```diff
@@ -0,0 +1,35 @@
+
+
+def _appendErrors(element: GBOMLObject, defNodeScope: DefNodeScope) -> None:
+    
+    # if element.path[0].name not in defNodeScope.content and element.path[0].name not in defNodeScope.parent.content:
+        # print(f"SEMANTIC ERROR: {element.path[0].name} not declared in this scope {element.meta}!")
+        # print(defNodeScope.parent)
+    # else:
+        # print(f"semantic ok: {element.path[0].name} declared in scope {element.meta}")
+
+    for subElement in element.path:
+        print(f"SubElement {subElement.name}")
+        # if isinstance(defNodeScope, RootScope):
+            # print(defNodeScope)
+            # break
+        if subElement.name not in defNodeScope.content and subElement.name not in defNodeScope.parent.content:
+            print(f"SEMANTIC ERROR: {subElement.name} not declared in this scope {subElement.meta}!")
+            print(defNodeScope.parent)
+        else:
+            print(f"semantic ok: {subElement.name} declared in scope {subElement.meta}")
+            # print(defNodeScope.parent)
+            if subElement.name in defNodeScope.parent.content:
+                defNodeScope = defNodeScope.parent.content[subElement.name]
+                print(defNodeScope)
+
+def _visit_nodes(nodeDef: NodeDefinition) -> None:
+    visit(nodeDef, {VarOrParam: lambda _: _appendErrors(_, nodeDef.scope)})
```

Comment:
In Python typically `_` is reserved for unused variables; here it is used so you should probably name it